crates/zksync/core/src/vm/tracer.rs

```diff
@@ -227,6 +241,29 @@ impl<S, H: HistoryMode> DynTracer<S, SimpleMemory<H>> for CheatcodeTracer {
                     return;
                 }
             }
+
+            let contract_code = Self::sload(storage, get_code_key(&current.code_address));
+            let is_empty_code = contract_code == hash_bytecode(&EMPTY_CODE) ||
+                contract_code == StorageValue::zero();
+
+            // if we get here there was no matching mock call,
+            // so we check if there's no code at the mocked address
+            if !self.is_code_expected_empty(call_contract) && is_empty_code {
+                // issue a more targeted
+                // error if we already had some mocks there
+                let had_mocks_message = if mocks.is_some() {
+                    " - please ensure the current calldata is mocked"
+                } else {
+                    ""
+                };
+
+                tracing::error!(
+                    target = ?call_contract,
+                    calldata = hex::encode(&call_input),
+                    "call may fail or behave unexpectedly due to empty code{}",
+                    had_mocks_message
+                );
+            }
```

Comment:
I believe this functionality is better represented by having a well encapsulated `fn check_empty_code_call(storage, target)` which also inlines the known addresses via a variable

```rs
// The following addresses are expected to have empty bytecode
let ignored_known_addresses = vec![
  foundry_common::HARDHAT_CONSOLE_ADDRESS.
  self.call_context.tx_caller,
]
```